### ISSUE TITLE
Error on saving Orderable models 

### ISSUE THREAD

**Issue by bartromgens** [jonge-democraten/website#112]
I get an error when I try to add a Orderable model to a page. 
Examples are a Document, SocialMediaButton, and ColumnElementWidget.

Steps to reproduce:
1. Edit a sidebar
2. Add a social media button
3. Save
4. Error

**Comment by Pieter1024** [comment]
I can confirm this error in commits as old as 7abeb733ee76edd1c207bc07e9c7b47038ed9d8d (have not looked any further). This would imply that this has stopped working before May. I find that odd: we've all played around enough with the functionality after that.

It could also be the case that one of our dependencies actually broke this functionality.

**Comment by bartromgens** [comment]
I also suspect that a dependency, one without a strict version defined, broke this functionality.

**Comment by Pieter1024** [comment]
As soon as this works, I will implement automated migration of ColumnElements and SidebarWidgets.

**Comment by Pieter1024** [comment]
I have tracked down the problem (with a little help from Stephen McDonald, obviously: https://groups.google.com/forum/#!searchin/mezzanine-users/invalid$20literal/mezzanine-users/9_cRBwrnxTc/GABVs-OLaU0J).

The error arises because an Orderable is saved without an order. If I run a local proxy and change the 'columnelementwidget_set-0-_order' value in the request parameters from '' to '0', no error arises and the element is added successfully.

Stephen suggests this could be the case because of JavaScript problems (i.e. JS should set the order but doesnt). A wild guess would be that our version of jQuery is really quite old and could be giving us some trouble here.

**Comment by Pieter1024** [comment]
My main suspect right now is /static/mezzanine/js/admin/dynamic_inline.js, because it states that its purpose is to set the order of fields that are submitted.

I have found out the following:

* In the following line, the orderSelector is determined. Since grappelli is installed, its value is always '._order input'. https://github.com/stephenmcd/mezzanine/blob/master/mezzanine/core/static/mezzanine/js/admin/dynamic_inline.js#L41

* In the following line, the orderSelector is used to select a number of entries for which to set the _order value. If I add a ColumnElement, the code below it is never executed (and thus no _order values are added). https://github.com/stephenmcd/mezzanine/blob/master/mezzanine/core/static/mezzanine/js/admin/dynamic_inline.js#L56

* I have tried to change the first line to something that would match all fields, but that only led to all fields (even the hidden ones) being validated. That was no use as well.

**Comment by Pieter1024** [comment]
Since I was able to programmatically import the ColumnElements through the migration scripts, this obviously is a frontend problem.

**Comment by bartromgens** [comment]
Thank you for the hints. 
It was a newer version of grappelli-safe that broke it. Mezzanine only requires a version >= 0.3.12, so 0.4.1 was used in new environments. Also see https://github.com/stephenmcd/mezzanine/issues/1289.

Fixed by forcing to use version 0.3.13 in requirements.txt. 

Run ``` $ pip install -r requirements.txt ``` to make this fix active.

**Comment by Pieter1024** [comment]
Thank you for looking into this. I still experience the same problem with your fixed version. I've pulled the latest commit to the test server and run a full ./migrate.sh. I still get the 'error' message.

Could you look into this?

**Comment by bartromgens** [comment]
Did you upgrade the virtual environment such that grappelli-safe v0.3.13 is used?

**Comment by Pieter1024** [comment]
I did a clean_env and then build_env. In the output, I saw grappelli-safe 0.3.13 being installed:

```
$ ./migrate.sh
+===================================================+
| 1) Setting up virtual env                         |
+===================================================+
rm: cannot remove ‘website/dev.db’: No such file or directory
Running virtualenv with interpreter /usr/bin/python3.4
Using base prefix '/usr'
New python executable in env/bin/python3.4
Also creating executable in env/bin/python
Installing setuptools, pip...done.
Downloading/unpacking django==1.7.8 (from -r requirements.txt (line 1))
  Using download cache from /tmp/pip_cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2Fpy2.py3%2FD%2FDjango%2FDjango-1.7.8-py2.py3-none-any.whl
Downloading/unpacking mezzanine==3.1.10 (from -r requirements.txt (line 2))
  Using download cache from /tmp/pip_cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2F2.7%2FM%2FMezzanine%2FMezzanine-3.1.10-py2.py3-none-any.whl
Downloading/unpacking grappelli-safe==0.3.13 (from -r requirements.txt (line 3))
  Using download cache from /tmp/pip_cache/https%3A%2F%2Fpypi.python.org%2Fpackages%2F2.7%2Fg%2Fgrappelli_safe%2Fgrappelli_safe-0.3.13-py2.py3-none-any.whl
```

**Comment by Pieter1024** [comment]
Oddly enough, I don't have the same problem on my local instance.

**Comment by bartromgens** [comment]
Looks like there is a different problem. I will look into this. Let me know if this blocks your work.

**Comment by Pieter1024** [comment]
Well, it is as bad as it used to be before your work, so it doesn't block it any further :).

**Comment by bartromgens** [comment]
I cannot reproduce this anymore on the test server. Seems to work as intended. 
@Pieter1024, can you verify it works now?

**Comment by Pieter1024** [comment]
I still have the same problem on the test server.

Steps to reproduce:
1. Edit the homepage of the main site
2. Add a new column element in the left column with title 'Extra' and element 'Weblog Headlines'.
3. Click 'Save and continue editing'.
4. -> Error

**Comment by Pieter1024** [comment]
This is odd: while I still have the problem with ColumnElementWidgets, I no longer have the problem with SidebarBlogCategoryWidgets. I assume that is the case you tested as well, which would explain the discrepancy between our experiences.

**Comment by bartromgens** [comment]
I tested exactly those steps, for the column widget. 

Just tried again. Works fine for me in Firefox. Tried several times and always works. 
Then tried with Chromium and got the error again. 
Then on a Windows Firefox: error.
Back to Firefox on Linux: works fine.

I am confused. 

The error from the Django logs is the same as before downgrading to grappelli-safe v0.3.13: "ValueError: invalid literal for int() with base 10: ''
